tf_encrypted/tensor/int100.py

```diff
@@ -379,6 +380,10 @@ def truncate(self, amount, base=2):
     def right_shift(self, bitlength):
         return self.truncate(bitlength, 2)
 
+    def cast(self, factory):
```

Comment:
This is saying int100tensors don't need to be casted so just return ourself?

Reply:
the assertion makes sure that you can only cast int100 it to int100; it's a bit ugly and should be cleanup up once we have more advanced types for tensors!

Reply:
for now it's needed to use the same code for securenn on int64 and int100 (where the former is cast to the odd tensor)